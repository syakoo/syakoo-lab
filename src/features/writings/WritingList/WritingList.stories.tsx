import type { Meta, StoryObj } from "@storybook/react";
import { WritingList } from ".";

const meta = {
  component: WritingList,
  parameters: {
    layout: "fullscreen",
  },
} satisfies Meta<typeof WritingList>;

export default meta;

type Story = StoryObj<typeof meta>;
export const Sample: Story = {
  args: {
    type: "all",
  },
};
