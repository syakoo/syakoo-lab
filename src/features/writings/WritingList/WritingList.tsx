"use client";

import { useSearchParams } from "next/navigation";
import { WritingBlock } from "./WritingBlock";
import { WritingTab } from "./WritingTab";
import { WritingTypeDescription } from "./WritingTypeDescription";
import type { WritingListType } from "./_shared/writingListType";
import { Col } from "@/design-system/layout";
import { H2, FadeIn } from "@/design-system/ui";
import type { WritingMeta } from "@/features/writings/types";

type WritingListProps = {
  metas: WritingMeta[];
};

export const WritingList: React.FC<WritingListProps> = ({ metas }) => {
  const searchParams = useSearchParams();
  const type = (searchParams.get("type") ?? "all") as WritingListType;

  const filteredMetas = metas.filter((meta) => {
    if (type === "all") return true;
    return type === meta.type;
  });

  return (
    <section>
      <Col gap="300">
        <Col gap="100">
          <H2>Writings</H2>
          <WritingTab selectedType={type} />
          <WritingTypeDescription type={type} />
        </Col>
        <Col key={type} as="ul" gap="200">
          {filteredMetas.map((meta, i) => (
            <FadeIn key={meta.id} as="li" delaySec={0.05 * i}>
              <WritingBlock meta={meta} />
            </FadeIn>
          ))}
        </Col>
      </Col>
    </section>
  );
};
