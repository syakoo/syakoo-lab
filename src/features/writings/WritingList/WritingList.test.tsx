import { testStories } from "@/test/testStories";

import * as stories from "./WritingList.stories";

describe("WritingList", () => {
  testStories(stories);
});
