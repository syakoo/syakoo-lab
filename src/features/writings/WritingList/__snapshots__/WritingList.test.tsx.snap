// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WritingList renders Sample 1`] = `
<body>
  <div>
    <section>
      <div
        class="Flex_flexStyle__15ivi84i Flex_flexStyle_align_stretch__15ivi84u Flex_flexStyle_variants_align_stretch__15ivi849 Flex_flexStyle_direction_column__15ivi84k Flex_flexStyle_gap_300__15ivi8410 Flex_flexStyle_variants_gap_300__15ivi84f Flex_flexStyle_justify_flexStart__15ivi84l Flex_flexStyle_variants_justify_flexStart__15ivi840"
      >
        <div
          class="Flex_flexStyle__15ivi84i Flex_flexStyle_align_stretch__15ivi84u Flex_flexStyle_variants_align_stretch__15ivi849 Flex_flexStyle_direction_column__15ivi84k Flex_flexStyle_gap_100__15ivi84y Flex_flexStyle_variants_gap_100__15ivi84d Flex_flexStyle_justify_flexStart__15ivi84l Flex_flexStyle_variants_justify_flexStart__15ivi840"
        >
          <h2
            class="Text_textStyle__jzpfqde Text_textStyle_color_currentColor__jzpfqdi Text_textStyle_variants_color_currentColor__jzpfqd3 Text_textStyle_size_500__jzpfqdp Text_textStyle_variants_size_500__jzpfqda Text_textStyle_weight_bold__jzpfqdt"
          >
            Writings
          </h2>
          <div
            class="Flex_flexStyle__15ivi84i Flex_flexStyle_align_stretch__15ivi84u Flex_flexStyle_variants_align_stretch__15ivi849 Flex_flexStyle_direction_row__15ivi84j Flex_flexStyle_gap_100__15ivi84y Flex_flexStyle_variants_gap_100__15ivi84d Flex_flexStyle_justify_flexStart__15ivi84l Flex_flexStyle_variants_justify_flexStart__15ivi840"
          >
            <a
              class="Link_linkStyle__m4v9430 Link_linkStyle_noHovered_true__m4v9433"
              href="/writings"
            >
              <div
                aria-current="true"
                class="WritingTab_writingTabStyle_item__1s7we910 WritingTab_writingTabStyle_item_active_true__1s7we911"
              >
                All
              </div>
            </a>
            <a
              class="Link_linkStyle__m4v9430"
              href="/writings?type=article"
            >
              <div
                aria-current="false"
                class="WritingTab_writingTabStyle_item__1s7we910 WritingTab_writingTabStyle_item_active_false__1s7we912"
              >
                Article
              </div>
            </a>
            <a
              class="Link_linkStyle__m4v9430"
              href="/writings?type=note"
            >
              <div
                aria-current="false"
                class="WritingTab_writingTabStyle_item__1s7we910 WritingTab_writingTabStyle_item_active_false__1s7we912"
              >
                Note
              </div>
            </a>
            <a
              class="Link_linkStyle__m4v9430"
              href="/writings?type=diary"
            >
              <div
                aria-current="false"
                class="WritingTab_writingTabStyle_item__1s7we910 WritingTab_writingTabStyle_item_active_false__1s7we912"
              >
                Diary
              </div>
            </a>
          </div>
        </div>
        <div
          class="Flex_flexStyle__15ivi84i Flex_flexStyle_align_stretch__15ivi84u Flex_flexStyle_variants_align_stretch__15ivi849 Flex_flexStyle_direction_column__15ivi84k Flex_flexStyle_gap_200__15ivi84z Flex_flexStyle_variants_gap_200__15ivi84e Flex_flexStyle_justify_flexStart__15ivi84l Flex_flexStyle_variants_justify_flexStart__15ivi840"
        >
          <article
            class="Flex_flexStyle__15ivi84i Flex_flexStyle_align_center__15ivi84s Flex_flexStyle_variants_align_center__15ivi847 Flex_flexStyle_direction_row__15ivi84j Flex_flexStyle_gap_200__15ivi84z Flex_flexStyle_variants_gap_200__15ivi84e Flex_flexStyle_justify_flexStart__15ivi84l Flex_flexStyle_variants_justify_flexStart__15ivi840"
          >
            <div
              class="WritingBlock_writingBlockStyles_iconWrapper__1bhf5f50"
            >
              <svg
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M19.5 14.25v-2.625a3.375 3.375 0 00-3.375-3.375h-1.5A1.125 1.125 0 0113.5 7.125v-1.5a3.375 3.375 0 00-3.375-3.375H8.25m0 12.75h7.5m-7.5 3H12M10.5 2.25H5.625c-.621 0-1.125.504-1.125 1.125v17.25c0 .621.504 1.125 1.125 1.125h12.75c.621 0 1.125-.504 1.125-1.125V11.25a9 9 0 00-9-9z"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </div>
            <div>
              <a
                class="Link_linkStyle__m4v9430"
                href="/writing/post/20231009"
              >
                <h3
                  class="Text_textStyle__jzpfqde Text_textStyle_color_currentColor__jzpfqdi Text_textStyle_variants_color_currentColor__jzpfqd3 Text_textStyle_size_200__jzpfqdm Text_textStyle_variants_size_200__jzpfqd7 Text_textStyle_weight_bold__jzpfqdt"
                >
                  Sample Article
                </h3>
              </a>
              <p
                class="Text_textStyle__jzpfqde Text_textStyle_color_secondary__jzpfqdg Text_textStyle_variants_color_secondary__jzpfqd1 Text_textStyle_size_50__jzpfqdj Text_textStyle_variants_size_50__jzpfqd4 Text_textStyle_weight_normal__jzpfqdu"
              >
                2023-10-09
              </p>
              <div
                class="Flex_flexStyle__15ivi84i Flex_flexStyle_align_stretch__15ivi84u Flex_flexStyle_variants_align_stretch__15ivi849 Flex_flexStyle_direction_row__15ivi84j Flex_flexStyle_gap_50__15ivi84x Flex_flexStyle_variants_gap_50__15ivi84c Flex_flexStyle_justify_flexStart__15ivi84l Flex_flexStyle_variants_justify_flexStart__15ivi840"
              >
                <p
                  class="Text_textStyle__jzpfqde Text_textStyle_color_secondary__jzpfqdg Text_textStyle_variants_color_secondary__jzpfqd1 Text_textStyle_size_50__jzpfqdj Text_textStyle_variants_size_50__jzpfqd4 Text_textStyle_weight_normal__jzpfqdu"
                >
                  #
                  雑記
                </p>
                <p
                  class="Text_textStyle__jzpfqde Text_textStyle_color_secondary__jzpfqdg Text_textStyle_variants_color_secondary__jzpfqd1 Text_textStyle_size_50__jzpfqdj Text_textStyle_variants_size_50__jzpfqd4 Text_textStyle_weight_normal__jzpfqdu"
                >
                  #
                  Sample
                </p>
              </div>
            </div>
          </article>
          <article
            class="Flex_flexStyle__15ivi84i Flex_flexStyle_align_center__15ivi84s Flex_flexStyle_variants_align_center__15ivi847 Flex_flexStyle_direction_row__15ivi84j Flex_flexStyle_gap_200__15ivi84z Flex_flexStyle_variants_gap_200__15ivi84e Flex_flexStyle_justify_flexStart__15ivi84l Flex_flexStyle_variants_justify_flexStart__15ivi840"
          >
            <div
              class="WritingBlock_writingBlockStyles_iconWrapper__1bhf5f50"
            >
              <svg
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16.862 4.487l1.687-1.688a1.875 1.875 0 112.652 2.652L10.582 16.07a4.5 4.5 0 01-1.897 1.13L6 18l.8-2.685a4.5 4.5 0 011.13-1.897l8.932-8.931zm0 0L19.5 7.125M18 14v4.75A2.25 2.25 0 0115.75 21H5.25A2.25 2.25 0 013 18.75V8.25A2.25 2.25 0 015.25 6H10"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </div>
            <div>
              <a
                class="Link_linkStyle__m4v9430"
                href="/writing/post/20231010"
              >
                <h3
                  class="Text_textStyle__jzpfqde Text_textStyle_color_currentColor__jzpfqdi Text_textStyle_variants_color_currentColor__jzpfqd3 Text_textStyle_size_200__jzpfqdm Text_textStyle_variants_size_200__jzpfqd7 Text_textStyle_weight_bold__jzpfqdt"
                >
                  Sample Note
                </h3>
              </a>
              <p
                class="Text_textStyle__jzpfqde Text_textStyle_color_secondary__jzpfqdg Text_textStyle_variants_color_secondary__jzpfqd1 Text_textStyle_size_50__jzpfqdj Text_textStyle_variants_size_50__jzpfqd4 Text_textStyle_weight_normal__jzpfqdu"
              >
                2023-10-10
              </p>
              <div
                class="Flex_flexStyle__15ivi84i Flex_flexStyle_align_stretch__15ivi84u Flex_flexStyle_variants_align_stretch__15ivi849 Flex_flexStyle_direction_row__15ivi84j Flex_flexStyle_gap_50__15ivi84x Flex_flexStyle_variants_gap_50__15ivi84c Flex_flexStyle_justify_flexStart__15ivi84l Flex_flexStyle_variants_justify_flexStart__15ivi840"
              >
                <p
                  class="Text_textStyle__jzpfqde Text_textStyle_color_secondary__jzpfqdg Text_textStyle_variants_color_secondary__jzpfqd1 Text_textStyle_size_50__jzpfqdj Text_textStyle_variants_size_50__jzpfqd4 Text_textStyle_weight_normal__jzpfqdu"
                >
                  #
                  雑記
                </p>
                <p
                  class="Text_textStyle__jzpfqde Text_textStyle_color_secondary__jzpfqdg Text_textStyle_variants_color_secondary__jzpfqd1 Text_textStyle_size_50__jzpfqdj Text_textStyle_variants_size_50__jzpfqd4 Text_textStyle_weight_normal__jzpfqdu"
                >
                  #
                  Sample
                </p>
              </div>
            </div>
          </article>
          <article
            class="Flex_flexStyle__15ivi84i Flex_flexStyle_align_center__15ivi84s Flex_flexStyle_variants_align_center__15ivi847 Flex_flexStyle_direction_row__15ivi84j Flex_flexStyle_gap_200__15ivi84z Flex_flexStyle_variants_gap_200__15ivi84e Flex_flexStyle_justify_flexStart__15ivi84l Flex_flexStyle_variants_justify_flexStart__15ivi840"
          >
            <div
              class="WritingBlock_writingBlockStyles_iconWrapper__1bhf5f50"
            >
              <svg
                fill="none"
                stroke="currentColor"
                stroke-width="1.5"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 6.042A8.967 8.967 0 006 3.75c-1.052 0-2.062.18-3 .512v14.25A8.987 8.987 0 016 18c2.305 0 4.408.867 6 2.292m0-14.25a8.966 8.966 0 016-2.292c1.052 0 2.062.18 3 .512v14.25A8.987 8.987 0 0018 18a8.967 8.967 0 00-6 2.292m0-14.25v14.25"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                />
              </svg>
            </div>
            <div>
              <a
                class="Link_linkStyle__m4v9430"
                href="/writing/post/20231011"
              >
                <h3
                  class="Text_textStyle__jzpfqde Text_textStyle_color_currentColor__jzpfqdi Text_textStyle_variants_color_currentColor__jzpfqd3 Text_textStyle_size_200__jzpfqdm Text_textStyle_variants_size_200__jzpfqd7 Text_textStyle_weight_bold__jzpfqdt"
                >
                  Sample Diary
                </h3>
              </a>
              <p
                class="Text_textStyle__jzpfqde Text_textStyle_color_secondary__jzpfqdg Text_textStyle_variants_color_secondary__jzpfqd1 Text_textStyle_size_50__jzpfqdj Text_textStyle_variants_size_50__jzpfqd4 Text_textStyle_weight_normal__jzpfqdu"
              >
                2023-10-11
              </p>
              <div
                class="Flex_flexStyle__15ivi84i Flex_flexStyle_align_stretch__15ivi84u Flex_flexStyle_variants_align_stretch__15ivi849 Flex_flexStyle_direction_row__15ivi84j Flex_flexStyle_gap_50__15ivi84x Flex_flexStyle_variants_gap_50__15ivi84c Flex_flexStyle_justify_flexStart__15ivi84l Flex_flexStyle_variants_justify_flexStart__15ivi840"
              >
                <p
                  class="Text_textStyle__jzpfqde Text_textStyle_color_secondary__jzpfqdg Text_textStyle_variants_color_secondary__jzpfqd1 Text_textStyle_size_50__jzpfqdj Text_textStyle_variants_size_50__jzpfqd4 Text_textStyle_weight_normal__jzpfqdu"
                >
                  #
                  雑記
                </p>
              </div>
            </div>
          </article>
        </div>
      </div>
    </section>
  </div>
</body>
`;
